ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -72,87 +82,273 @@ def test_h_max_results():
+    (using TelescopeFrame) provide compatible results and that we are able to
+    reconstruct a positive number of events.
+    """
+
+    filename = get_dataset_path(
+        "gamma_LaPalma_baseline_20Zd_180Az_prod3b_test.simtel.gz"
+    )
 
     source = EventSource(filename, max_events=10)
     calib = CameraCalibrator(subarray=source.subarray)
 
-    horizon_frame = AltAz()
+    fit_CameraFrame = HillasReconstructor(source.subarray)
+    fit_TelescopeFrame = HillasReconstructor(source.subarray)
 
     reconstructed_events = 0
 
+    # ==========================================================================
+
     for event in source:
+
         calib(event)
 
-        mc = event.simulation.shower
-        array_pointing = SkyCoord(az=mc.az, alt=mc.alt, frame=horizon_frame)
-        hillas_dict = {}
+        # make a copy of the event for the camera frame case
+        event_CameraFrame = deepcopy(event)
+
+        hillas_dict_CameraFrame = {}
+        hillas_dict_TelescopeFrame = {}
         telescope_pointings = {}
+
         for tel_id, dl1 in event.dl1.tel.items():
 
-            geom = source.subarray.tel[tel_id].camera.geometry
+            # this is needed only here to transform the camera geometries
+            telescope_pointings[tel_id] = SkyCoord(
+                alt=event.pointing.tel[tel_id].altitude,
+                az=event.pointing.tel[tel_id].azimuth,
+                frame=AltAz(),
+            )
+
+            geom_CameraFrame = source.subarray.tel[tel_id].camera.geometry
+
+            # this could be done also out of this loop,
+            # but in case of real data each telescope would have a
+            # different telescope_pointing
+            geom_TelescopeFrame = geom_CameraFrame.transform_to(
+                TelescopeFrame(telescope_pointing=telescope_pointings[tel_id])
+            )
+
+            mask = tailcuts_clean(
+                geom_TelescopeFrame, dl1.image, picture_thresh=10.0, boundary_thresh=5.0
+            )
+
+            try:
+                moments_CameraFrame = hillas_parameters(
+                    geom_CameraFrame[mask], dl1.image[mask]
+                )
+                moments_TelescopeFrame = hillas_parameters(
+                    geom_TelescopeFrame[mask], dl1.image[mask]
+                )
+                hillas_dict_CameraFrame[tel_id] = moments_CameraFrame
+                hillas_dict_TelescopeFrame[tel_id] = moments_TelescopeFrame
+
+                event_CameraFrame.dl1.tel[
+                    tel_id
+                ].parameters = ImageParametersContainer()
+                dl1.parameters = ImageParametersContainer()
+
+                event_CameraFrame.dl1.tel[
+                    tel_id
+                ].parameters.hillas = moments_CameraFrame
+                dl1.parameters.hillas = moments_TelescopeFrame
+
+                if (
+                    (moments_CameraFrame.width.value == np.nan)
```

Comment:
Something is off here.
I assume it should be `moments_TelescopeFrame` two times

Reply:
and you cannot compare to nan. You need to use `np.isnan`

Reply:
fixed